.github/workflows/test.yaml

```diff
@@ -12,14 +14,28 @@ jobs:
   testing:
     runs-on: ubuntu-20.04
     env:
-      TOXENV: "alldeps-withcov-posix,coverage-prepare,codecov-push,coveralls-push"
+      TOXENV: "${{ matrix.tox-env }}-withcov-posix,coverage-prepare,codecov-push,coveralls-push"
+    name: ${{ matrix.python-version }}${{ matrix.noipv6 }}-${{ matrix.tox-env }}-withcov-posix
     strategy:
       matrix:
-          # Run on the minimum micro Python version that we can get on CI.
-          # When updating the minimum Python version here, also update the
-          # `python_requires` from `setup.cfg`.
-          # Run on latest minor release of each major python version.
-          python-version: ["3.6.7", 3.6, 3.7, 3.8, 3.9, pypy-3.6, pypy-3.7]
+        # Run on the minimum micro Python version that we can get on CI.
+        # When updating the minimum Python version here, also update the
+        # `python_requires` from `setup.cfg`.
+        # Run on latest minor release of each major python version.
+        python-version: ["3.6.7", 3.8, 3.9, pypy-3.6, pypy-3.7]
```

Comment:
I prefer 'complete' matrices...  but I haven't been dealing with Twisted lately so I don't know how painful the builds are.  For what is being done, it seems to be done well.